src/nodes/kolibri-broker/kolibri-broker.ts

```diff
@@ -0,0 +1,333 @@
+/*
+* Copyright 2021 HMS Industrial Networks AB
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http: //www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+import { ClientConfig, KolibriClient, KolibriClientFactory, LoginParams, SubscribeParams, SubscribeResult, UnsubscribeParams, WriteParams } from "@hms-networks/kolibri-js-client";
+import { NodeInitializer, Node } from "node-red";
+import { IKolibriBrokerNode, KolibriBrokerNodeDef, Subscription } from "./modules/types";
+
+const nodeInit: NodeInitializer = (RED): void => {
+  class KolibriBrokerNode implements IKolibriBrokerNode {
+    name: string;
+    broker: string;
+    port: number;
+    project: string;
+    brokerUrl: string;
+    connected: boolean;
+    connecting: boolean;
+    closing: boolean;
+    user: string;
+    password: string;
+    nodes: Map<string, Node>;
+    subscriptions: Map<string, Subscription>;
+    client: KolibriClient;
+    useProxy: boolean;
+    proxyHost: string;
+    proxyPort: number;
+    proxyProtocol: string;
+    private lazyThis: Node<{ user: string, password: string }> & IKolibriBrokerNode;;
+    constructor(config: KolibriBrokerNodeDef) {
+      this.lazyThis = this as unknown as Node<{ user: string, password: string }> & IKolibriBrokerNode;
```

Comment:
Why are we creating a copy of this with this type? Isnt the KolibriBrokerNode object already a IKolibriBrokerNode and Node ?

Reply:
This is a shortcoming from the static types of typescript. On the class definition phase we don't know that this will be also a Node Type after the registration on runtime. 
We could also implement everything from the Node interface, but this would be to much boilerplate code.